halo2/src/prover.rs

```diff
@@ -219,32 +206,19 @@ impl Halo2Prover {
         Ok(proof)
     }
 
-    pub fn assert_field_is_compatible<F: FieldElement>() {
-        if polyexen::expr::get_field_p::<Fr>() != F::modulus().to_arbitrary_integer() {
-            panic!("powdr modulus doesn't match halo2 modulus. Make sure you are using Bn254");
-        }
-    }
-
-    fn verification_key_inner<F: FieldElement>(
-        &self,
-        pil: &Analyzed<F>,
-        fixed: &[(String, Vec<F>)],
-    ) -> Result<VerifyingKey<G1Affine>, String> {
-        let plaf_circuit = analyzed_to_plaf(pil, fixed);
-        let circuit = analyzed_to_circuit_with_zeroed_witness(plaf_circuit, pil);
-        match keygen_vk(&self.params, &circuit) {
-            Ok(vkey) => Ok(vkey),
-            Err(e) => Err(e.to_string()),
-        }
+    pub fn add_verification_key(&mut self, mut vkey: &mut dyn io::Read) {
```

Comment:
I guess we don't need this function anymore?

Reply:
As I said previously, I think `powdr_halo2` belongs as a mod inside backend crate, but I didn't want to do any any deeper change to the `Halo2Prover` in this PR.